packages/app/src/components/ExpandableRowTable.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { memo, useCallback, useState } from 'react';
+import cx from 'classnames';
+import { useQueryState } from 'nuqs';
+import { TSource } from '@hyperdx/common-utils/dist/types';
+
+import { useLocalStorage } from '@/utils';
+
+import TabBar from '../TabBar';
+
+import { RowDataPanel } from './DBRowDataPanel';
+import { RowOverviewPanel } from './DBRowOverviewPanel';
+
+import styles from '../../styles/LogTable.module.scss';
+
+enum InlineTab {
+  Overview = 'overview',
+  ColumnValues = 'columnValues',
+}
+
+// Hook that provides a function to open the sidebar with specific row details
+const useSidebarOpener = () => {
+  const [, setRowId] = useQueryState('rowWhere');
+  const [, setRowSource] = useQueryState('rowSource');
+
+  return useCallback(
+    (rowWhere: string, sourceId: string) => {
+      setRowId(rowWhere);
+      setRowSource(sourceId);
+    },
+    [setRowId, setRowSource],
+  );
+};
+
+export const ExpandedLogRow = memo(
+  ({
+    columnsLength,
+    virtualKey,
+    source,
+    rowId,
+  }: {
+    columnsLength: number;
+    virtualKey: string;
+    source: TSource | undefined;
```

Comment:
no need for a change, my preference is that potentially reusable components take ids instead of objects, and fetch dependencies they need themselves. Makes it easier to reuse across the app without the parent needing to do extra data fetching.

i know this is more of an internal compnent to the row table for now, so not necessarily releveant, but perhaps possible this turns into a reused one in a custom dashboard or something one day.